src/pkgs.jl

```diff
@@ -94,7 +94,7 @@ function pkg_fileinfo(id::PkgId)
     # Try to find the matching cache file
     paths = Base.find_all_in_cache_path(id)
     sourcepath = Base.locate_package(id)
-    if length(paths) > 1
+    if sourcepath !== nothing && length(paths) > 1
```

Comment:
If `sourcepath === nothing` is there ever the possibility that `length(paths) > 1`? And if so, is there any risk we could grab the wrong `path`?

Reply:
I tweaked it slightly to make it behave more similar to the normal case. I am not 100% sure what the code here does, I just know that it doesn't handle the case where `locate_package` returns `nothing` which is a possibility as the issue above shows.